node/directives/checkAdminAccess.ts

```diff
@@ -20,9 +24,21 @@ export class CheckAdminAccess extends SchemaDirectiveVisitor {
         vtex: { adminUserAuthToken, storeUserAuthToken, logger },
       } = context
 
-      const { hasAdminToken, hasValidAdminToken, hasCurrentValidAdminToken } =
+      let { hasAdminToken, hasValidAdminToken, hasCurrentValidAdminToken } =
         await validateAdminToken(context, adminUserAuthToken as string)
 
+      let hasAdminTokenOnHeader = false
+
+      // If there's no admin token on context, search for it on header
+      if (!hasAdminToken) {
+        ;({
+          hasAdminToken,
+          hasValidAdminToken,
+          hasCurrentValidAdminToken,
+          hasAdminTokenOnHeader,
```

Comment:
Instead of reusing the variables from the `validateAdminToken` result, what do you think about creating new `hasAdminTokenOnHeader`, `hasValidAdminTokenOnHeader`, etc? This way metrics will be more intuitive to understand.

Reply:
Done